README.md

```diff
@@ -18,13 +18,60 @@
 The goal of this project is to handle most of the Swarm operations through CLI at some point in the future.
 For currently supported operations, see [Commands](##Commands) section.
 
+## Installation
+
+### From npm
+
+```sh
+ $ npm install -g @ethersphere/swarm-cli
+```
+
+### From source
+
+See the [Development](##Development) section.
+
+## Usage
+
+The general usage is to provide a command, optionally a subcommand, then arguments and options.
+
+`swarm-cli command [subcommand] <arguments> [options]`
+
+Running a command with the `--help` option prints out the usage of a command.
+
 ## Commands
 
-`swarm-cli` can perform commands:
-- upload # upload files (even directories) to Swarm network by passing the file's path
-- identity # manage keys (which can be compatible with Ethereum V3 keystore standard) that you can use mostly for signing chunks
+Running `swarm-cli` without arguments prints the available commands:
+
+```
+$ swarm-cli
+swarm-cli <command>
+
+Commands:
+  index.js upload <path>  - Upload file to Swarm
+  index.js identity       - Keypair management interface
+  index.js feed           - Feed utilities
+
+```
+
+## Example usage
+
+Let's say we want to upload a directory of our website to Swarm and update a feed to point to the newest version. For updating a feed we would need to sign it with an Ethereum key, so first we need to create one with the `identity create` command:
+
+```
+swarm-cli identity create
+```
 
-for more info execute `swarm-cli --help` after installation.
+This command will ask for a password. After that a new identity is created (named `main`). Now we can use this identity to sign updates. It's also possible to import and export Ethereum JSON V3 format identities that works with other apps (e.g. wallets).
+
+For uploading to a feed we can use the `feed upload` command. It expects an `identity` to be provided along with the `password` belonging to it and the `path` of the directory (or file) we want to upload.
```

Comment:
Directories are files
```suggestion
For uploading to a feed we can use the `feed upload` command. It expects an `identity` to be provided along with the `password` that belongs to it and the `path` of the file (even of a directory) we want to upload.
```

Reply:
I'm not sure I understand what you mean, can you please clarify?

Please note that the example is about uploading a website which will be most likely a directory. Also I plan to change directory to folder everywhere to be more consistent with the user expectations too.